Rooms basic typeclass with flags and sectors handling

time- opened the pull request.

Commit:
Overloaded return_appearance method for autoexits and better formatting. To be checked

Commit:
Added room wikipage

Commit:
initial flag work

Commit:
Merge branch 'master' of https://github.com/evennia/ainneve into rooms

Commit:
Updated room typeclass, added @flag command

Commit:
better formatting help string

Commit:
minox fixes and wiki update

Commit:
Merge branch 'master' of https://github.com/evennia/ainneve

Commit:
Merge remote-tracking branch 'origin/master' into rooms

Commit:
added room_size property

Commit:
implemented Nadorrano's FlagHandler

Commit:
did some code cleaning and fixed .lists method

Commit:
Added sectors code and sectors handler

Commit:
removed Urban class

whitenoiseoss commented:
Most of this is not necessary, unfortunately. This is done in a very DIKU style, which is an anti-pattern for Evennia development. The details can be found in my mailing list response here: https://groups.google.com/forum/#!category-topic/evennia/ainneve/wjboQ9qXy_k

whitenoiseoss commented:
I am going to close this at the moment.

ergodicbreak commented:
Reopening as I think it makes more sense for time- to modify this PR in place, and until we know we will not merge this PR I don't see a reason to close.

Commit:
fixed room typeclass - added long/shortdescs

Commit:
Merge branch 'master' of https://github.com/evennia/ainneve into rooms

time- commented:
removed unused attributes, added baseobject.py with a BaseObject class to inherit from. added long and short desc as property (good for dynamic changes as nadorrano proposed) and touched object/character typeclasses to inherit from baseobject also.
return_appearance room method is now fixed. it can do a lot with minor effort.
hope this is fine :)

whitenoiseoss commented on typeclasses/characters.py:
```diff
@@ -15,8 +15,10 @@
 
-class Character(DefaultCharacter):
+
+class Character(BaseObject, DefaultCharacter):
```
This multiple inheritance should not be happening.

whitenoiseoss commented on typeclasses/objects.py:
```diff
@@ -11,8 +11,9 @@
+from baseobject import BaseObject
 
-class Object(DefaultObject):
+class Object(BaseObject, DefaultObject):
```
This multiple inheritance should not be happening.

whitenoiseoss commented on typeclasses/rooms.py:
```diff
@@ -6,9 +6,11 @@
 
-
-class Room(DefaultRoom):
+class Room(BaseObject, DefaultRoom):
```
This multiple inheritance should not be happening.

whitenoiseoss commented on typeclasses/rooms.py:
```diff
@@ -18,4 +20,64 @@ class Room(DefaultRoom):
+    # return the item short_desc or some "in_room_desc" (ex. A sword is dropped
+    # here)
+    
+    def return_appearance(self, looker):
```
This is good for now.

whitenoiseoss commented on utils/flags_handler.py:
```diff
@@ -0,0 +1,58 @@
+class FlagsHandler(object):
```
This needs to be deleted.

whitenoiseoss commented on utils/sectors_handler.py:
```diff
@@ -0,0 +1,32 @@
+class SectorsHandler(object):
```
This should also be deleted for the moment, even if we are going to eventually implement terrain, "sector" is not the right word for it. In this implementation, I don't know that we are doing terrain negotiation.

whitenoiseoss commented on typeclasses/baseobject.py:
```diff
@@ -0,0 +1,19 @@
+from evennia import DefaultObject
```
These properties are technically not necessary, but they make other code prettier, so I'm fine with these properties existing. However, this should be using super so all of the below multiple inheritances are not necessary. Multiple inheritance should be very scarce.

whitenoiseoss replied:
In addition, we don't need "typeclasses/baseobject", we already have "typeclasses/objects.py" with an Object class in it. Thing should inherit from that typeclasses.objects.Object and not DefaultObject in Evennia.

time- replied:
Ok I did all the cuts, but I'm stuck here. I don't know how to set these properties on objects.Object, let all the typeclasses inherit from there, but without doing multiple inheritance. If I change the parent class, they won't get any of the defaults from the evennia code I guess, and super can be used only on direct parent class, right?
sorry mate I'm too noobie to solve this. For now I just doubled the code, adding properties into Room, Character and Item typeclass

whitenoiseoss commented on typeclasses/Rooms.md:
```diff
@@ -0,0 +1,52 @@
+# Room typeclass
+### Flags and Sectors
```
Everything in this file describes features that are either anti-pattern or not needed.

whitenoiseoss commented on commands/build.py:
```diff
@@ -0,0 +1,129 @@
+#    @property
```
This whole file should be removed.

whitenoiseoss commented on commands/default_cmdsets.py:
```diff
@@ -16,6 +16,7 @@
 
 from evennia import default_cmds
 from commands import equip_commands
+from commands import build
```
If the above whole file is removed, these lines must be removed as well.

whitenoiseoss commented on typeclasses/rooms.py:
```diff
@@ -18,4 +20,64 @@ class Room(DefaultRoom):
-    pass
+    
+    # Sectors handled by sectors handler
+    @property
```
This property should be removed.

whitenoiseoss commented on typeclasses/rooms.py:
```diff
@@ -18,4 +20,64 @@ class Room(DefaultRoom):
+        return SectorsHandler(self)
+    
+    # Flags handled by Nadorrano's handler
+    @property
```
This property should be removed.

Commit:
Fixed as per PR comments. Room.md removed as not needed anymore

whitenoiseoss commented on typeclasses/rooms.py:
```diff
@@ -7,7 +7,6 @@
 
 from evennia import DefaultRoom
```
Room should be pulling from typeclasses.object.Object, and then you don't need the property code anymore. All 3 of these could be consolidated with a proper use of super. DefaultCharacter inherits from DefaultObject and just does some extra stuff to it. Object should have the properties, and Character and Room can both pull from Object. Character will need its locks.

whitenoiseoss commented on typeclasses/characters.py:
```diff
@@ -40,7 +40,22 @@ class Character(DefaultCharacter):
 
     """
-
+    @property
```
See comment on Room for this. These properties should only be in one place.

Commit:
Fixed chars/room inheritance from objects.Object and removed unecessary properties

Commit:
Fixed terrains - removed old handlers
